.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,32 @@
+version: '3.8'
+
+services:
+  app:
+    build:
+      context: ..
+      dockerfile: .devcontainer/Dockerfile
+
+    volumes:
+      - ../..:/workspaces:cached
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:postgres
+
+  postgres:
+    image: postgres:latest
+    restart: always
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_DB: postgres
+      POSTGRES_PASSWORD: postgres
+    # Note that these ports are ignored by the devcontainer.
+    # Instead, the ports are specified in .devcontainer/devcontainer.json.
+    # ports:
+    #  - "5432:5432"
```

Comment:
I found this confusing, so I'd prefer keeping this comment in, but happy to remove if you think it's too much noise.